pkgs/by-name/ma/maskprocessor/package.nix

```diff
@@ -0,0 +1,39 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  versionCheckHook,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "maskprocessor";
+  version = "0.73";
+
+  src = fetchFromGitHub {
+    owner = "hashcat";
+    repo = "maskprocessor";
+    tag = "v${version}";
+    hash = "sha256-LVtMz5y0PbKKuc92W5xW0C84avigR7vS1XL/aXkUYe8=";
+  };
+
+  # upstream Makefile is terrible, this simplifies everything.
+  buildPhase = ''
+    $CC -o maskprocessor src/mp.c -W -Wall -std=c99 -O2 -s -m64 -DLINUX
+  '';
+
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  doInstallCheck = true;
+
+  installPhase = ''
+    install -Dm544 -t $out/bin maskprocessor
+  '';
+
+  meta = {
+    homepage = "https://github.com/hashcat/maskprocessor";
+    description = "High-Performance word generator with a per-position configureable charset.";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.unix;
+    changelog = "https://github.com/hashcat/maskprocessor/releases/tag/v${version}";
+    maintainers = with lib.maintainers; [ ethancedwards8 ];
```

Comment:
```suggestion
    maintainers = with lib.maintainers; [ ethancedwards8 ];
    mainProgram = "maskprocessor";
```

Reply:
Why? The binary is the same as the package name?

Reply:
Because the behavior of `lib.getExe` (https://github.com/NixOS/nixpkgs/pull/246386) was changed a while back. That was followed by https://github.com/NixOS/nixpkgs/pull/297084 which added `meta.mainProgram` to packages with a single binary.

Reply:
I see. Thanks for explaining.

Reply:
oops, yes.